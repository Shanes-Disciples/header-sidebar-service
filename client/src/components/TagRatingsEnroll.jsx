import React from 'react';

const TagRatingsEnroll = ({ data }) => (
  <section className="tag_ratings_enroll">
    <div className="tag">{ data.tag }</div>
    <div className="ratings">
      <div className="stars"></div>
      <div className="avg-rating">{ data.avg_rating }</div>
      <div className="total-ratings">{ `(${data.total_ratings} ratings)` }</div>
    </div>
    <div className="enrollment">{ `${data.enrollment} students enrolled` }</div>
  </section>
);

export default TagRatingsEnroll;