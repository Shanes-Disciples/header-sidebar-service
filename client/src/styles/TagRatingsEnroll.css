h3, .tagRatingsEnroll, .authorUpdate {
  width: 100%;
  padding-bottom: 8px;
}

.tagRatingsEnroll, .authorUpdate, .languagesContainer {
  font-weight: 200;
}

.tagRatingsEnroll {
  display: flex;
  justify-content: flex-start;
  flex-wrap: wrap;
}

.ratings {
  margin-right: 15px;
}

.tag {
  position: relative;
  color: #29303b;
  border-radius: 3px;
  padding: 3px 10px 3px 3px;
  z-index: 1;
  vertical-align: baseline;
  top: -2.6px;
  font-weight: 400;
  font-size: 0.9em;
}

.tag:after {
  content: "";
  background: inherit;
  border-radius: 3px;
  position: absolute;
  transform: rotate(45deg);
  z-index: -1;
  width: 18.5px;
  height: 16.7px;
  top: 2.35px;
  right: -8px;
  vertical-align: baseline;
}

.avgRating {
  padding-right: 2px;
}