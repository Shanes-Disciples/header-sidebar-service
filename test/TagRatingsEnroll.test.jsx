import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TagRatingsEnroll from '../client/src/components/TagRatingsEnroll.jsx';

configure({ adapter: new Adapter() });

const sampleProps = {
  active_coupon: 'ILOVEUDEMO',
  avg_rating: 4.9,
  ccOptions: ['English', 'Spanish'],
  created_by: 'Sibyl Roberts',
  description: 'Iste qui quia in.',
  discount_price: '$28.99',
  enrollment: 51,
  id: 1,
  img_url: 'http://lorempixel.com/640/480',
  language: 'English, Mandarin',
  last_updated: '21/2015',
  list_price: '$283.99',
  tag: 'Hot & New',
  title: 'React & Redux for Beginners',
  total_articles: 3,
  total_downloads: 5,
  total_ratings: 41,
  video_hrs: 26.1,
};

const samplePropsNull = {
  active_coupon: 'ILOVEUDEMO',
  avg_rating: 4.9,
  ccOptions: ['English', 'Spanish'],
  created_by: 'Sibyl Roberts',
  description: 'Iste qui quia in.',
  discount_price: '$28.99',
  enrollment: 51,
  id: 1,
  img_url: 'http://lorempixel.com/640/480',
  language: 'English, Mandarin',
  last_updated: '21/2015',
  list_price: '$283.99',
  tag: null,
  title: 'React & Redux for Beginners',
  total_articles: 3,
  total_downloads: 5,
  total_ratings: 41,
  video_hrs: 26.1,
};

describe('<TagRatingsEnroll />', () => {
  it('renders correctly', () => {
    const TagRatingsEnrollWrapper = renderer.create(
      <TagRatingsEnroll course={ sampleProps }/>,
    ).toJSON();
    expect(TagRatingsEnrollWrapper).toMatchSnapshot();
  });
  it('should render a div tag with class "tag" when tag is not null', () => {
    const wrapper = shallow(<TagRatingsEnroll course={ sampleProps }/>);
    expect(wrapper.exists('.tag')).toEqual(true);
  });
  it('should not render a div tag with class "tag" when tag is null', () => {
    const wrapper = shallow(<TagRatingsEnroll course={ samplePropsNull }/>);
    expect(wrapper.exists('.tag')).toEqual(false);
  });
});
