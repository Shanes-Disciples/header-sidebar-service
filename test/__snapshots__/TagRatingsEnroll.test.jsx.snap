// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagRatingsEnroll /> renders correctly 1`] = `
<section
  className="tagRatingsEnroll"
>
  <div
    className="tag"
    style={
      Object {
        "background": "#ec5252",
        "color": "#fff",
        "marginRight": "15px",
      }
    }
  >
    Hot & New
  </div>
  <div
    className="ratings"
  >
    <span>
      <span
        className="stars"
      >
        <span
          className="fullStar"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
        <span
          className="fullStar"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
        <span
          className="fullStar"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
        <span
          className="fullStar"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
        <span
          className="partialStar"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </span>
      <svg
        height="0"
        width="0"
      >
        <linearGradient
          id="starPartialFill"
        >
          <stop
            offset="0%"
            stopColor="#f4c150"
          />
          <stop
            offset="90%"
            stopColor="#f4c150"
          />
          <stop
            offset="90%"
            stopColor="#dedfe0"
          />
          <stop
            offset="100%"
            stopColor="#dedfe0"
          />
        </linearGradient>
        <linearGradient
          id="starEmpty"
        >
          <stop
            offset="100%"
            stopColor="#dedfe0"
          />
        </linearGradient>
        <linearGradient
          id="starFilled"
        >
          <stop
            offset="100%"
            stopColor="#f4c150"
          />
        </linearGradient>
      </svg>
    </span>
    <span
      className="avgRating"
    >
      4.9
    </span>
    <span>
      (41 ratings)
    </span>
  </div>
  <div>
    51 students enrolled
  </div>
</section>
`;
